vllm/distributed/device_communicators/custom_all_reduce.py

```diff
@@ -138,21 +138,10 @@ def __init__(self,
         else:
             device_ids = list(range(torch.cuda.device_count()))
 
-        physical_device_id = device_ids[device.index]
-        tensor = torch.tensor([physical_device_id],
-                              dtype=torch.int,
-                              device="cpu")
-        gather_list = [
-            torch.tensor([0], dtype=torch.int, device="cpu")
-            for _ in range(world_size)
-        ]
-        dist.all_gather(gather_list, tensor, group=self.group)
-        physical_device_ids = [t.item() for t in gather_list]
-
         # test nvlink first, this will filter out most of the cases
         # where custom allreduce is not supported
         # this checks hardware and driver support for NVLink
-        full_nvlink = _is_full_nvlink(physical_device_ids)
+        full_nvlink = _is_full_nvlink(device_ids)
```

Comment:
This is not true. Sometimes, e.g. in the testing code, we manually remove cuda visible devices variables, and the process can see all device ids. However, we only need to check nvlink topology within several gpus, rather than all.